components/script/dom/document.rs

```diff
@@ -2778,6 +2793,10 @@ impl Document {
                 );
                 event.set_trusted(true);
 
+                timing.update_load_event_start(start_time);
+                window.upcast::<EventTarget>().dispatch_event(&event, CanGc::note());
+                timing.update_load_event_end((*performance.Now()).floor() as u64);
```

Comment:
We already dispatch the event below in `window.dispatch_event_with_target_override(&event, CanGc::note());`, so we should update the timing information based on that event.